bash/bash_profile_vs_bashrc.md

```diff
@@ -0,0 +1,40 @@
+# `.bash_profile` vs `.bashrc`
+
+`.bashrc` is sourced on every interactive shell start. Here are some ways to
+start `bash(1)` interactively:
+
+    bash
+    bash -i
+    bash -ic 'echo Hello!'
+
+`bash(1)` is also started interactively by your terminal emulator.
+
+Here are some ways to start bash in non-interactive mode:
+
+    bash script.sh
+    bash -c 'echo Hello!'
+    bash --login
+
+`.bash_profile` is only sourced when `bash(1)` is started as an interactive
+login shell, or as a non-interactive shell with the `--login` option.
```

Comment:
Thoughts on linking to a source explaining the differences between interactive/non-interactive/login shells?